pydicom/dataset.py

```diff
@@ -358,6 +358,13 @@ def __getitem__(self, key):
                 character_set = default_encoding
             # Not converted from raw form read from file yet; do so now
             self[tag] = DataElement_from_raw(data_elem, character_set)
+
+            # If the Element has an ambiguous VR, try to correct it
+            if 'or' in self[tag].VR:
```

Comment:
perhaps make a utility function for `is_ambiguous` and then down the road this can be changed to not have to parse strings?